ocrd_models/ocrd_models/ocrd_exif.py

```diff
@@ -36,29 +39,27 @@ def __init__(self, img):
         #      print(img.info)
         for prop in ['compression', 'photometric_interpretation']:
             setattr(self, prop, img.info[prop] if prop in img.info else None)
-        if img.format in ('TIFF', 'PNG') and 'dpi' in img.info:
-            self.xResolution = int(img.info['dpi'][0])
-            self.yResolution = int(img.info['dpi'][1])
-            if img.format == 'TIFF':
-                self.resolutionUnit = 'cm' if img.tag.get(296) == 3 else 'inches'
-            else:
-                self.resolutionUnit = 'inches'
-        elif img.format == 'JPEG' and 'jfif_density' in img.info:
-            self.xResolution = img.info['jfif_density'][0]
-            self.yResolution = img.info['jfif_density'][1]
-            self.resolutionUnit = 'cm' if img.info['jfif_unit'] == 2 else 'inches'
-        elif img.format == 'PNG' and 'aspect' in img.info:
-            self.xResolution = img.info['aspect'][0]
-            self.yResolution = img.info['aspect'][1]
+        if img.format == 'JPEG2000':
+            # XXX find a way to get pixel density in jpeg2000
+            self.xResolution = self.yResolution = 1
```

Comment:
Well yes. Pillow does _open_ JP2 already, so ocrd_exif should go along here. Pillow does not yield `.info` for JP2 though, and IM only does after raising its [libopenjp2 requirement](https://bugs.launchpad.net/ubuntu/+source/imagemagick/+bug/1447968) – shipped in Ubuntu 20.10ff.

So, no: why not try your luck with the IM CLI below, and only _then_ fallback?

Reply:
I was not aware of this, good to hear. Once (if) we upgrade our base image, we can have proper JP2 support.

Reply:
Upgrade base image... that would require careful inspection of all ramifications. AFAICS we still have no solution for TF1 / Python 3.8 support ...

JP2 / JB2: already discussed at length in #526 BTW

Reply:
True, but I hope things will imporve in the next years.

Reply:
OCR-D Tech Call: Ask OCR-D Community if they have more cases for testing